### PR TITLE
feat: cmd added to stop the application

### PR DESCRIPTION
**Pull request by ndsrivatsa**
Stop application  Issue #117 : https://github.com/ios-control/ios-deploy/issues/117

### CODE REVIEW

File: src/ios-deploy/ios-deploy.m [hunk 1]
```diff
@@ -50,11 +50,18 @@
 const char* lldb_prep_noninteractive_justlaunch_cmds = "\
     run\n\
     safequit\n\
+    kill\n\
+";
+
+const char* lldb_prep_stopapp_cmds = "\
+kill\n\
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: Use the same whitespace convention as found in the cmds immediately above and below the change.

File: src/ios-deploy/ios-deploy.m [hunk 2]
```diff
@@ -1823,7 +1834,7 @@ int main(int argc, char *argv[]) {
     };
     int ch;
 
-    while ((ch = getopt_long(argc, argv, "VmcdvunNrILeD:R:i:b:a:s:t:g:x:p:1:2:o:l::w::9::B::W", longopts, NULL)) != -1)
+     while ((ch = getopt_long(argc, argv, "VSmcdvunrILeD:R:i:b:a:t:g:x:p:1:2:o:l::w::9::B::W", longopts, NULL)) != -1)
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: You broke the -s option when adding your -S option

File: src/ios-deploy/ios-deploy.m [hunk 3]
```diff
@@ -729,7 +737,9 @@ void write_lldb_prep_cmds(AMDeviceRef device, CFURLRef disk_app_url) {
     const char* extra_cmds;
     if (!interactive)
     {
-        if (justlaunch)
+        if (stopapp)
+        extra_cmds = lldb_prep_stopapp_cmds;
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: Indent the line correctly

File: src/ios-deploy/ios-deploy.m [hunk 4]
```diff
@@ -1845,6 +1856,11 @@ int main(int argc, char *argv[]) {
         case 's':
             envs = optarg;
             break;
+        case 'S':
+            stopapp = true;
+            justlaunch = true;
+            install = false;
+            debug = false;
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: Missing break;

File: src/ios-deploy/ios-deploy.m [hunk 5]
```diff
@@ -50,11 +50,18 @@
 const char* lldb_prep_noninteractive_justlaunch_cmds = "\
     run\n\
     safequit\n\
+    kill\n\
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: Why did you add an additional command here?

File: src/ios-deploy/ios-deploy.m [hunk 6]
```diff
@@ -50,11 +50,18 @@
...
+
+const char* lldb_prep_stopapp_cmds = "\
+kill\n\
+autoexit\n\
 ";
 
 const char* lldb_prep_noninteractive_cmds = "\
     run\n\
     autoexit\n\
+    kill\n\
```
Comment (palmerc) [inline src/ios-deploy/ios-deploy.m]: You added kill to this command as well?

**Comment by ndsrivatsa** [comment]
I have not cleaned up the code. Was done in a hurry. Cleaning up in 30 min and will commit the update.

File: src/ios-deploy/ios-deploy.m (hunk 1 above)
Comment (ndsrivatsa) [reply src/ios-deploy/ios-deploy.m]: fixed

File: src/ios-deploy/ios-deploy.m (hunk 2 above)
Comment (ndsrivatsa) [reply src/ios-deploy/ios-deploy.m]: fixed

File: src/ios-deploy/ios-deploy.m (hunk 6 above)
Comment (ndsrivatsa) [reply src/ios-deploy/ios-deploy.m]: fixed

**Comment by iprox** [comment]
Hi @ndsrivatsa thanks for the pull request! I've checked out the branch and tried to use the command, but it doesn't seem to work for me.
```
> ios-deploy -S
```
it gives me an error
```
[ !! ] One of -[b|c|o|l|w|D|R|e|9] is required to proceed!
```
tried to use it with other flabs including `-I` & `-b` but nothing seems to work for me
how is the new flag is supposed to be used? thanks

**Comment by palmerc** [comment]
I was wondering the same thing. Shouldn't the command take a bundle id?

**Comment by shazron** [comment]
What's the status of this PR? The PR says it has conflicts now. Usually it's simple to resolve via the Github UI, but GH is saying its too complex to resolve. I'll have to move this PR off the 1.10.0  milestone for now.

**Comment by gabebear** [comment]
I'm closing this for now... while there is some utility in stopping an app I'm not sure who really needs it and this PR has been open for a year.

**Comment by david-strauss-privoro** [comment]
Hi @gabebear ,

Currently using iOS-Deploy for some automated integration-level testing support and have a need to be able to fully stop the app from the command line. Can we re-open this and please add support to stop the app?

**Comment by gabebear** [comment]
We can reopen this PR if @ndsrivatsa wants to or you can create a new PR based off this work.

It's possible your use-case would fit into https://github.com/ios-control/ios-deploy/issues/169 better. Killing the debugger will also terminate the process/app it is connected to.

**Comment by david-strauss-privoro** [comment]
Thanks @gabebear I'll check that out!

**Comment by gabebear** [comment]
We don't have a simple solution to this issue right now. You _could_ implement all the lldb python scripting yourself and get this functionality, but that is difficult.

I don't know if anyone is actively working on this right now... I left #169 open as what I believe is the core of the problem here.
